### DISCUSSION TITLE
Issues Using Linked Singin option with supabase OAuth

### DISCUSSION THREAD

**Discussion by Daniel-Alamezie** [supabase/supabase#22711] [Questions]
I am setting up a project that uses linkedIn signin options with supabase when i enter the user email and password it redirects me to my homepage with this message.
Anyone have ideas why this could be ?
?error=server_error&error_code=500&error_description=Error+getting+user+profile+from+external+provider#error=server_error&error_code=500&error_description=Error+getting+user+profile+from+external+provider.

for context this is what my client and server code looks like.

## Client code
export default function LoginPage() {
  useEffect(() => {
    if (window.location.search.includes('error')) {
      console.error('Error during the OAuth flow');
    }
  }, []);

  async function signInWithLinkedIn() {
    const supabase = supabaseBrowser();
    const { error } = await supabase.auth.signInWithOAuth({
      provider: 'linkedin_oidc',
      options: {
        redirectTo: `${window.location.origin}/auth/callback`,
      }
    });
    if (error) {
      console.error('Error signing in:', error);
    }
  }

 return (
   <Button variant="outline" className="w-full" onClick={signInWithLinkedIn}>
              Login with LinkedIn
            </Button>
)

## server code
import { cookies } from 'next/headers'
import { NextResponse } from 'next/server'
import { type CookieOptions, createServerClient } from '@supabase/ssr'

export async function GET(request: Request) {
  const { searchParams, origin } = new URL(request.url)
  const code = searchParams.get('code')
  // if "next" is in param, use it as the redirect URL
  const next = searchParams.get('next') ?? '/'

  if (code) {
    const cookieStore = cookies()
    const supabase = createServerClient(
      process.env.NEXT_PUBLIC_SUPABASE_URL!,
      process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
      {
        cookies: {
          get(name: string) {
            return cookieStore.get(name)?.value
          },
          set(name: string, value: string, options: CookieOptions) {
            cookieStore.set({ name, value, ...options })
          },
          remove(name: string, options: CookieOptions) {
            cookieStore.delete({ name, ...options })
          },
        },
      }
    )
    const { error } = await supabase.auth.exchangeCodeForSession(code)
    if (!error) {
      return NextResponse.redirect(`${origin}/`)
    }
  }

  // return the user to an error page with instructions
  return NextResponse.redirect(`${origin}/auth/auth-code-error`)
}

**Comment by GaryAustin1** [comment]
https://github.com/orgs/supabase/discussions/22708
Seems like several reports of similar things.  One on Discord too, but could be one of you two.

**Reply by GaryAustin1** [reply]
Seems a user generated an issue and proposes a fix:  https://github.com/supabase/auth/pull/1534

**Reply by Daniel-Alamezie** [reply]
Ah so a fix has already been done, just waiting to be merged?

**Reply by GaryAustin1** [reply]
Well, a user generated the fix... so if that user is correct on what the issue is and the fix...

**Reply by Daniel-Alamezie** [reply]
okay thanks

**Reply by kangmingtay** [reply]
hey everyone, seems like this was a breaking change that wasn't being communicated on linkedin's end :( - we've made a [fix](https://github.com/supabase/auth/pull/1536) for it but it will take awhile before we are able to roll it out to everyone on the hosted platform 

if you need this urgently, please open a ticket at https://supabase.help

**Reply by Daniel-Alamezie** [reply]
Thanks for the update.
